### PR TITLE
Hotfix for #1356

### LINKED ISSUE #1356: Creating a new note crashes the app
**Describe the bug**

The app crashes when I open a book and click on **New Note**.

**Expected behavior**

The app should not crash and **New Note** should be renamed to **Notes** and moved down the menu.

**Steps to reproduce the behavior:**
1. Open a book.
1. Go to menu -> New Note

**Environment**
- Version of Kiwix Android : develop
- Device : Redmi Note 3
- OS version : Android M

**Logs**
```
java.lang.StringIndexOutOfBoundsException: length=110; regionStart=43; regionLength=-8
        at java.lang.String.startEndAndLength(String.java:298)
        at java.lang.String.substring(String.java:1087)
        at org.kiwix.kiwixmobile.main.AddNoteDialog.getTextAfterLastSlashWithoutExtension(AddNoteDialog.java:205)
        at org.kiwix.kiwixmobile.main.AddNoteDialog.getArticleNotefileName(AddNoteDialog.java:188)
        at org.kiwix.kiwixmobile.main.AddNoteDialog.onCreate(AddNoteDialog.java:105)
        at androidx.fragment.app.Fragment.performCreate(Fragment.java:2414)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1418)
```

### PR DESCRIPTION
**Pull request by Aditya-Sood**
Fixes #1356 

Changes: Added check for extension-less article urls and 'null' articles (ones rendered erroneously without any content)

### CODE REVIEW

**Comment by Aditya-Sood** [comment]
@abdulwd it works with the main page of ArchWiki now, does it work for other specific articles that were crashing?

**Comment by Aditya-Sood** [comment]
@julianharty if possible could you please check the changes with android-monkey?

**Comment by Aditya-Sood** [comment]
@macgills I'm not sure whether [this line](https://github.com/kiwix/kiwix-android/compare/hotfix/1356#diff-36e13fa4a9132616aee81e1f88bfa0ffR191) is really necessary. Should I be worried about what's being returned even if the whole dialog is getting closed?

**Comment by julianharty** [comment]
> @julianharty if possible could you please check the changes with android-monkey?

I will have a go at testing this with robot. I have made a copy of the repo and checked out your branch hotfix/1356 I'll let you know tomorrow (Saturday in Europe) how my testing with Android Monkey goes.

**Comment by abdulwd** [comment]
I'll let you know tomorrow.

File: app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java [hunk 1]
```diff
@@ -106,15 +102,19 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
...
-      closeKeyboard();
-      getFragmentManager().beginTransaction().remove(AddNoteDialog.this).commit();
+      onFailureToCreateAddNoteDialog();
     }
   }
 
+  private void onFailureToCreateAddNoteDialog() {
+    showToast(R.string.error_filenotfound, Toast.LENGTH_LONG);
+    closeKeyboard();
+    getFragmentManager().beginTransaction().remove(AddNoteDialog.this).commit();
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java]: you don't need to qualify `this` with `AddNoteDIalog`

Comment (Aditya-Sood) [reply app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java]: [Done](https://github.com/kiwix/kiwix-android/pull/1387/commits/06fcfa55d5643c6baf0204422505fb087db95cb6#diff-36e13fa4a9132616aee81e1f88bfa0ffR112)

**Comment by Aditya-Sood** [comment]
@macgills I'm not sure whether [this line](https://github.com/kiwix/kiwix-android/compare/hotfix/1356#diff-36e13fa4a9132616aee81e1f88bfa0ffR191) is really necessary. Should I be worried about what's being returned even if the whole dialog is getting closed?

**Comment by macgills** [comment]
@Aditya-Sood yeah you can delete the return statement

**Comment by macgills** [comment]
@abdulwd I believe you mean `inspections`, if the build is green `lint` has passed (and also every commit is linted).
If we should be treating inspection more seriously then I believe we should raise their level in AS. All developers now share inspections in version control so we can mandate what inspections are important to the project, I have raised all available kotlin `inspections` to error for example so they get red highlighting

**Comment by abdulwd** [comment]
@macgills I think we should treat inspections more seriously and the pre-commit hook should fail if they're not resolved.

**Comment by macgills** [comment]
@abdulwd unfortunately extremely difficult, running inspections is far far harder. We can ask contributors to commit via AS 
image
and ask that the correct boxes are ticked? That is about as good as it gets

**Comment by abdulwd** [comment]
Yes, we can do this. We should also ask them to tick reformat code, rearrange code and optimize. Lastly, they should import our code style or should we add that file to git?

**Comment by macgills** [comment]
@abdulwd codestyle already added to git. I also don't think that requiring studio to commit files is the best way forward, people rarely read the contributing at the best of times.
This problem disappears the more we use kotlin as ktlint is integrated and should forbid improperly formatted code (and autoformat when you attempt to commit unstyled code), detekt will also be integrated in the future.
Regardless this PR is probably not the place to discuss this as I am about to merge this. To talk about our options going forward could you open a ticket?

**Pull request merged**